openldap/datadog_checks/openldap/openldap.py

```diff
@@ -0,0 +1,248 @@
+# (C) Datadog, Inc. 2018
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from copy import copy
+import ldap3
+import os
+import ssl
+
+from datadog_checks.checks import AgentCheck
+from datadog_checks.errors import CheckException
+from datadog_checks.config import is_affirmative
+
+
+SEARCH_BASE = "cn=Monitor"
+SEARCH_FILTER = "(objectClass=*)"
+ATTRS = ["*", "+"]
+
+METRIC_PREFIX = "openldap"
+
+# Some docs here https://www.openldap.org/doc/admin24/monitoringslapd.html#Monitor%20Information
+CONNECTIONS_METRICS_DN = "cn=connections,cn=monitor"
+OPERATIONS_METRICS_DN = "cn=operations,cn=monitor"
+STATISTICS_METRICS_DN = "cn=statistics,cn=monitor"
+THREADS_METRICS_DN = "cn=threads,cn=monitor"
+TIME_METRICS_DN = "cn=time,cn=monitor"
+WAITERS_METRICS_DN = "cn=waiters,cn=monitor"
+
+
+class OpenLDAP(AgentCheck):
+    def check(self, instance):
+        url, username, password, ssl_params, custom_queries, tags = self._get_instance_params(instance)
+
+        server = ldap3.Server(url, tls=self._get_tls_object(ssl_params))
+        conn = ldap3.Connection(server, username, password, collect_usage=True)
+
+        # Try binding to the server
+        try:
+            res = conn.bind()
+            if not res:
+                raise ldap3.core.exceptions.LDAPBindError("Error binding to server: {}".format(conn.result))
+        except ldap3.core.exceptions.LDAPBindError:
```

Comment:
Let's catch `as e`, put it in the log string, then return instead of raise.

Reply:
Don't we want the check to fail so we can see it in `status` ?

Reply:
Tbh, I don't know we have a standard. :slightly_smiling_face: